### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: conditionals

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1012]
This document describes how to implement the `conditionals` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student about the various `conditionals` within Common Lisp. The student should be able to recognize which conditional is the most appropriate in a given context and be able to confidently write `when`, `unless`, `if`, `cond`, and `case` expressions.

Additionally, the student will be introduced to the concept of "true" and "false" values. That is to say, in Common Lisp, `nil` and `'()` are "false", while all other values are "true"

## Learning objectives

- Know how to use `when` and `unless` for single-branch conditions
- Know how to use `if` for two-branch conditions
- Know how to use `cond` for multi-branch conditions
- Know how to use `case` and `ecase` for special, multi-branch conditions
- Understand which values are "false" and which are "true"

## Out of scope

- Sameness / Equality (`eq`, `eql`, `equal`, `equalp`, etc.)
- Short-circuit evaluation with `and` and `or`
- Conditions and restarts – those are *very* different ;)
- The `ccase` macro, which actually does involve conditions and restarts
- The type-matching `typecase` macro
- Any non-conditional or compound form of control flow

## Concepts

- `conditionals`
- `truthy-and-falsy`

## Prerequisites

- `expressions`
- `integers`
- `arithmetic`
- `symbols`

## Resources to refer to

- [Riptutorial](https://riptutorial.com/common-lisp/example/11082/conditional-constructs)
- [Common Lisp the Language](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node84.html)

### Hints

- TBD

### After

- TBD

## Representer

- Currently, no extra logic needs to be added to the [representer](https://github.com/exercism/common-lisp-representer).

## Analyzer

- The [analyzer](https://github.com/exercism/common-lisp-analyzer) would benefit from the ability to check which conditionals are employed by the student. It's often not possible to test for `cond` vs `if` vs `when`, so the analyzer, not the test suite, needs to pick up on this.

## Implementing

When implementing this exercise, avoid the topic of `sameness` as much as possible, as that is its own concept. Stick to only using values as true / false or basic numerical relations (`<`, `>=`, `=`, etc.) and avoid `eq`, `eql`, `equal`, etc.

The `case` macro is a bit of an exception here. Just be sure to use symbols or numbers here so that the default `:test` does not need to be changed from `eql`.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by verdammelt** [comment]
Re-reviewing this and have some questions: is the idea of a `sameness` concept still on the table?(and characters could be safe to use in `case` as well...).

I can't think of how to get a student to use `when` and `unless` however. That is how to write a test that will cause a student to use `when` rather than `if` or for that matter `cond`. It seems that the differences are stylistic rather than functional. Perhaps it will be the job of the analyzer/representer to help steer the student? But will those be in play for concept exercises? (i'll have to look into that).

**Comment by TheLostLambda** [comment]
I'm thinking that `sameness` could still be it's own concept, as that seems like a sizeable chunk of information. And we could use characters if that's easier, I just hadn't added it to avoid adding `characters` as a prerequisite.

I think that the analysers will be active for concept exercises and, as you point out, this is probably to perfect place to apply one. I'll add a note in the issue under the analyser section!
